### ISSUE TITLE
Incorrect time per step estimation when fitting model 

### ISSUE THREAD

**Issue by koalive** [keras-team/keras#16286]
Hi there,

**System information**.
I observed the behavior both in a Colab notebook (TF v2.8.0-0-g3f878cff5b6 2.8.0) and in a custom Docker image (Ubuntu 20.04, TF v2.5.1-97-g957590ea15c 2.5.2). The issue can easily be reproduced by using validation sets of different sizes and I made an example notebook (see below).

**Describe the problem**.
The time/step reported when calling `fit` is not the time per training step. The phrasing can be misleading, for instance when trying to design your training scheme based on time constraints. 

**Describe the current behavior**.
Currently, the time for a full epoch (including validation) divided by the number of training steps is reported. If the validation takes a significant amount of time, the time per training step might be way smaller than reported.

**Describe the expected behavior**.
I feel that reporting the time per training step (excluding validation) would be more informative.
For instance for the following output (from the Colab notebook linked below):
```
Epoch 1/3
5/5 [==============================] - 2s 613ms/step - loss: 0.3212 - val_loss: 0.1224
```
I would expect that using a single step instead of 5 would make each epoch take around 613ms. However each epoch would still take 2s as most of it is spent on the validation set.

**Standalone code to reproduce the issue**.

See this colab notebook:
https://colab.research.google.com/drive/1YGWstYcbFwkPY4ezZ2C-krDPahS36nnm?usp=sharing

Cheers.

**Comment by fig666** [comment]
i understand what your implying    when it comes to giving simpler  yet efficient movement in a small amount of time instead of trying to validate the move itself which giving my full focus on the movement for dynamics

**Comment by sachinprasadhs** [comment]
This is most unlikely scenario in general use case and currently `time/step` which is averaging out the time taken per step of training data and validation data makes sense.
